### ISSUE TITLE
Allow changing ConnectivityManagers mid-game.

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#2852]
kOS should be changed to tolerate switching ConnectivityManagers without having to start a brand new save.  Right now the user is only presented with the choice of ConnectivityManager at campaign start and then it's locked in from then on.

The reason for locking in the choice originally was because that was done in the long-ago before-times when KSP didn't let mods have their own settings panels on the difficulty options screen yet, and this was seen as a "rule choice" for the campaign, not a day-to-day change.  But now that difficulty options can be changed mid-game, it no longer fits to have this *one* choice be locked in at the start when none of the others are.  Also, kOS *can* tolerate changing the setting by editing the save file so we already know it's not going to cause bugs.  It's just that there's no user interface for doing it.

Another reason to change it is that In issue #2851, I made the comment that it's possible to install RT after the campaign save was started.  That would mean the Remote tech connectivity manager couldn't be selected when the campaign started but could be now.

Possible gotcha to check for:

* Because the setting could previously only be changed mid-game by editing the text files, that meant that technically we've only proven you can change it *between runs* of the KSP exe, not that you can change it *mid-run*.  It's possible that either (1) Allowing you to changing the setting mid-game may require clearing and recreating some things, or (2) We may need a note warning the player their change can't take effect until they quit and re-run the game.

About point (1) above, the question arises "What if some command is already enqueued in the delay buffer, but then you change the setting so it ignores signal delay and no longer has a delay buffer of commands?"  Probably the right answer is to just wipe the buffered commands and throw them out, since you're getting rid of the buffer system entirely anyway.

**Comment by captain828** [comment]
First of all, when a new save is created without RT installed, the choices are All or CommNet.
The user is asked about this on the first load with kOS installed.
The settings page allows changing between the two currently available connectivity managers.

When RT is installed, on the next load the user is told that RT was detected and asks if it should switch the ConnectivityManager to RT.
Furthermore, the RT manager now becomes available in the settings page.

Changing settings mid-save (tested on career mode) works as mentioned in this enhancement:
- the user can switch from all to commNet or RT whenever he pleases
- the selected connectivity manager is properly persisted in the sfs save file

Can't comment on the buffer system as I have no proper way of testing this.
I would consider it an edge case to be honest as it would mean switching from RT to All or CommNet, which implies a permanent loss of RT settings (antenna targets) if the user were to switch back and forth.

To conclude, this enhancement already exists.
It might require more specific testing (mod list, actual steps taken on UI etc.) if someone reports this issue.

**Comment by Dunbaratu** [comment]
It's still useful to let people change their mind.

**Comment by captain828** [comment]
Yeah, not saying it shouldn't be allowed just saying that I don't think it's worth the coding effort for an edge case (from RT to All or CommNet with commands in the queue).
